### PR TITLE
Update to Chromium 92.0.4515.131

### LINKED ISSUE #1561: Updating to Chromium  92
Chromium 92 will came in 1-2 weeks (maybe jul 20th)  

So ... could we   prepare for chromium 92  
taking the actual 92.0.4515.80   as base

### PR DESCRIPTION
**Pull request by Ahrotahn**
This PR updates Chromium to 92.0.4515.107.  

<br>

Notable changes:

`fix-crash-in-ThemeService.patch` has been removed now that the changes have finally landed.  

`fix-building-with-prunned-binaries.patch` has been updated to remove the change in [2849065](https://chromium-review.googlesource.com/c/chromium/src/+/2849065) since strip-eu has been prunned.  

Crashpad was enabled on linux in [2822475](https://chromium-review.googlesource.com/c/chromium/src/+/2822475), but since we've had crash reporting disabled I've updated `disable-crash-reporter.patch` to make `IsCrashpadEnabled()` return false.  

`disable-privacy-sandbox.patch` is a new patch that disables the privacy sandbox.  

The code to remotely disable extensions has been split off from safebrowsing and into OmahaAttributesHandler.  I've put the change in the safebrowsing removal patch for now, but that may need to be split off into it's own patch in the future.  

<br>

[Blog post](https://chromereleases.googleblog.com/2021/07/stable-channel-update-for-desktop_20.html)
[Change log](https://chromium.googlesource.com/chromium/src/+log/91.0.4472.164..92.0.4515.107?pretty=fuller&n=10000)

Fixes #1561

### CODE REVIEW

**Comment by PF4Public** [comment]
Looking good:
image

**Comment by networkException** [comment]
Runs fine for me as well :+1:

**Comment by Area51Kacz** [comment]
testing
https://build.opensuse.org/package/live_build_log/home:Area51Kacz:branches:home:ungoogled_chromium:testing/ungoogled-chromium-arch333/Arch/x86_64

**Comment by wchen342** [comment]
I'm getting
```
ld.lld: error: undefined symbol: quarantine::mojom::Quarantine::Name_
>>> referenced by interface_ptr_state.h:249 (../../mojo/public/cpp/bindings/lib/interface_ptr_state.h:249)
>>>               obj/content/browser/browser/file_system_access_file_writer_impl.o:(mojo::internal::InterfacePtrState<quarantine::mojom::Quarantine>::ConfigureProxyIfNecessary())
>>> referenced by interface_ptr_state.h:249 (../../mojo/public/cpp/bindings/lib/interface_ptr_state.h:249)
>>>               obj/content/browser/browser/file_system_access_file_writer_impl.o:(mojo::internal::InterfacePtrState<quarantine::mojom::Quarantine>::ConfigureProxyIfNecessary())

ld.lld: error: undefined symbol: vtable for quarantine::mojom::QuarantineResponseValidator
>>> referenced by quarantine.mojom.h:145 (gen/components/services/quarantine/public/mojom/quarantine.mojom.h:145)
>>>               obj/content/browser/browser/file_system_access_file_writer_impl.o:(std::__Cr::__unique_if<quarantine::mojom::QuarantineResponseValidator>::__unique_single std::__Cr::make_unique<quarantine::mojom::QuarantineResponseValidator>())
>>> referenced by quarantine.mojom.h:145 (gen/components/services/quarantine/public/mojom/quarantine.mojom.h:145)
>>>               obj/content/browser/browser/file_system_access_file_writer_impl.o:(std::__Cr::__unique_if<quarantine::mojom::QuarantineResponseValidator>::__unique_single std::__Cr::make_unique<quarantine::mojom::QuarantineResponseValidator>())
>>> the vtable symbol may be undefined because the class is missing its key function (see https://lld.llvm.org/missingkeyfunction)

ld.lld: error: undefined symbol: quarantine::mojom::QuarantineProxy::QuarantineProxy(mojo::MessageReceiverWithResponder*)
>>> referenced by memory:2006 (../../buildtools/third_party/libc++/trunk/include/memory:2006)
>>>               obj/content/browser/browser/file_system_access_file_writer_impl.o:(std::__Cr::__unique_if<quarantine::mojom::QuarantineProxy>::__unique_single std::__Cr::make_unique<quarantine::mojom::QuarantineProxy, mojo::InterfaceEndpointClient*>(mojo::InterfaceEndpointClient*&&))
clang-12: error: linker command failed with exit code 1 (use -v to see invocation)
```

on Android, it seems `patches/extra/ungoogled-chromium/disable-download-quarantine.patch` are missing some files.

**Comment by Ahrotahn** [comment]
That was fast!  It looks good to me and doesn't cause any issues building or running so far.  

I think `disable-download-quarantine.patch` was added to extra since it still built and ran without the patch, whereas this change would be needed to compete the build.  It makes more sense to include it with the existing changes to `file_system_access_file_writer_impl` in `fix-building-without-safebrowsing.patch`.  I'll make a new commit adding that in a bit.

**Comment by Zoraver** [comment]
I am going to have to disagree regarding adding the quarantine-specific changes to `fix-building-without-safebrowsing.patch`. Unless quarantine depends on safe browsing in some way, I think that these changes should go in `disable-download-quarantine.patch`. Consider users that do not want to disable the mark of the web; they build without `disable-download-quarantine.patch`, but download quarantine will be broken.

**Comment by Ahrotahn** [comment]
You're right, for some reason I was under the impression that quarantine was already being skipped due to safebrowsing's removal.  I'll move that change over into `disable-download-quarantine.patch` in a bit.

**Comment by Ekaterina-Lebedew** [comment]
could anyone provide a 64bit windows 10 installer ? i tried to build a version myself but.. well.. im too dumb.. :(

**Comment by kramred** [comment]
Could someone please test the print feature in v92. I've tried to update the macOS patches and got a [finished build](https://github.com/kramred/ungoogled-chromium-macos/releases) but the print dialogue seems to be broken. In the terminal I get the following error messages:

```
ERROR:content_settings_pref.cc(286)] Invalid pattern strings: [*.].google.com,*
ERROR:content_settings_pref.cc(286)] Invalid pattern strings: https://[*.].slack.com,*
ERROR:ev_root_ca_metadata.cc(841)] Failed to register OID: 0
ERROR:CONSOLE(191)] "Uncaught Error: Unexpected condition on chrome://print/: Could not find value for forceEnablePrivetPrinting", source: chrome://resources/js/load_time_data.m.js (191)
ERROR:print_preview_handler.cc(837)] Initiated PrintPreviewHandler::HandleGetInitialSettings()
ERROR:CONSOLE(191)] "Uncaught (in promise) Error: Unexpected condition on chrome://print/: Could not find value for forceEnablePrivetPrinting", source: chrome://resources/js/load_time_data.m.js (191)
ERROR:CONSOLE(2435)] "Uncaught TypeError: Cannot read property 'resetTracker' of null", source: chrome://print/print_preview.js (2435)
```
The print dialogue/modal window keeps showing "Loading preview..." and "Destination" the blue spinner animation.

v92_print_modal_macOS

**Comment by networkException** [comment]
Can reproduce on Linux

**Comment by wchen342** [comment]
Not sure whether it is related but I also got a loading problem with [Bromite's proxy patch](https://github.com/bromite/bromite/blob/master/build/patches/Add-a-proxy-configuration-page.patch), which adds a `chrome://proxy` page but on my Debug build for cr92 it cannot get past loading screen.

**Comment by Ahrotahn** [comment]
The `forceEnablePrivetPrinting` boolean in `chrome/browser/ui/webui/print_preview/print_preview_ui.cc` is now surrounded with buildflags checking if service discovery is enabled.  Since we build with `enable_service_discovery=false` removing the checks solves the immediate issue.  

Once I'm done with lunch I'll dig into the commit the change was added in to be sure there aren't any other areas that impacts and I'll update with a fix afterwards.

**Comment by kramred** [comment]
Thanks @Ahrotahn, I can confirm that removing the buildflags results in a (dirty) build where the print dialogue works again.

v92_print_works

**Comment by Ahrotahn** [comment]
@wchen342 I took a look into the proxy patch, see if making this change to the patch works:
```diff
--- a/patches/Bromite/Add-a-proxy-configuration-page.patch
+++ b/patches/Bromite/Add-a-proxy-configuration-page.patch
@@ -344,6 +344,6 @@
 --- /dev/null
 +++ b/chrome/browser/resources/proxy_config.js
-@@ -0,0 +1,263 @@
+@@ -0,0 +1,266 @@
 +/*
 +    This file is part of Bromite.
 +
@@ -361,6 +361,8 @@
 +    along with Bromite. If not, see <https://www.gnu.org/licenses/>.
 +*/
 +
++import {addSingletonGetter} from 'chrome://resources/js/cr.m.js';
++
 +/**
 + * Main entry point called once the page has loaded.
 + */
@@ -418,7 +418,8 @@
 +    chrome.send('enableNotifyUIWithState');
 +  }
 +
-+  cr.addSingletonGetter(ProxyConfigView);
++  addSingletonGetter(ProxyConfigView);
++  window.ProxyConfigView = ProxyConfigView;
 +
 +  ProxyConfigView.prototype = {
 +    /**
```

I'd test it myself, but Sylvain Beucler's development kit builds are no longer available 😞

**Comment by wchen342** [comment]
@Ahrotahn Thanks for the fix. I will test it and see whether it fixed the problem.

> I'd test it myself, but Sylvain Beucler's development kit builds are no longer available disappointed

Yeah the sdk is supposedly backed by the f-droid community if I remembered correctly, but since some time ago the website was gone. Though normal SDK and NDK from Google will work too, I am using it only because of the terms of the SDK.

**Comment by wchen342** [comment]
I tested the change but unfortunately it doesn't work. But don't block on this, I will ask @csagan5 separately.

File: patches/extra/bromite/flag-fingerprinting-canvas-image-data-noise.patch [hunk 1]
```diff
@@ -65,29 +65,29 @@ approach to change color components.
...
            snapshot->PaintImageForCurrentFrame().GetSkImageInfo().bounds();
        DCHECK(!bounds.intersect(SkIRect::MakeXYWH(sx, sy, sw, sh)));
      }
 +    if (read_pixels_successful && RuntimeEnabledFeatures::FingerprintingCanvasImageDataNoiseEnabled()) {
-+      base::span<uint8_t> dst = base::span<uint8_t>(reinterpret_cast<uint8_t*>(contents.Data()),
-+          contents.DataLength());
-+      StaticBitmapImage::ShuffleSubchannelColorData(dst.data(), image_info, sx, sy);
++      base::span<uint8_t> dst = base::span<uint8_t>(reinterpret_cast<uint8_t*>(image_data_pixmap.writable_addr()),
++          image_data_pixmap.rowBytes());
++      StaticBitmapImage::ShuffleSubchannelColorData(dst.data(), image_data_pixmap.info(), sx, sy);
```
Comment (Zoraver) [inline patches/extra/bromite/flag-fingerprinting-canvas-image-data-noise.patch:83]: `Multiple-fingerprinting-mitigations.patch` from Bromite 92 has a simpler way of handling this.

**Comment by csagan5** [comment]
> @Ahrotahn Thanks for the fix. I will test it and see whether it fixed the problem.

For the records, this is the fix from @uazo and it works fine on Bromite.

> I tested the change but unfortunately it doesn't work. But don't block on this, I will ask @csagan5 separately.

The fix was provided by @uazo in https://github.com/bromite/bromite/pull/1291 and I manually ported it, verified that it works fine to open `chrome://proxy` both when typing the URL or using the settings page (Android).

**Comment by csagan5** [comment]
On a side topic, @uazo we could check if ungoogled-chromium's patch to disable safe browsing has improvements that we miss in Bromite, since the two have had a separate evolution for long time now.

**Comment by dot-gov** [comment]
>`disable-privacy-sandbox.patch` is a new patch that disables the privacy sandbox.

Is this a good thing or a bad thing? One of the features of the privacy sandbox is a "privacy budget", which tries to prevent sites from fingerprinting. Disabling it seems like a step backwards?

**Comment by Ahrotahn** [comment]
It's a good thing.  The privacy sandbox sounds beneficial on the surface, but raises many concerns once one digs into it.  FLoC has gotten a lot of media attention, the other components you can read about on the chromium site [here](https://www.chromium.org/Home/chromium-privacy/privacy-sandbox).  You can follow the link for privacy budget near the bottom to see how it works.  People that are privacy-aware would consider enabling telemetry for privacy to be a contradiction, especially when Google is involved.  

On a more practical note privacy budget isn't fully implemented and is currently not enabled.  We've also already disabled FLoC, so there isn't much reason to keep the privacy sandbox currently.

**Comment by PF4Public** [comment]
> You can follow the link for privacy budget near the bottom to see how it works.

Excerpt to put the discussion into the context:

> Measure information exposed by each fingerprinting surface
> Each browser can start by instrumenting itself to estimate the information revealed by each surface and report it back through telemetry. This data will be used for the next phase.
> 
> Measure total information exposed to each site
> This is yet another measurement step where each browser can use the data collected in the previous step to account for the total information exposed to each site and report that back through telemetry.

**Comment by Ahrotahn** [comment]
I've just added a new commit for the 92.0.4515.131 update.  

[Blog post](https://chromereleases.googleblog.com/2021/08/the-stable-channel-has-been-updated-to.html)
[Change log](https://chromium.googlesource.com/chromium/src/+log/92.0.4515.107..92.0.4515.131?pretty=fuller&n=10000)

**Comment by Area51Kacz** [comment]
Trying it

**Comment by PF4Public** [comment]
.131 compiled and running well

**Review by Eloston — APPROVED** [review]
LGTM, Thanks!

**Pull request merged**